docker/Dockerfile

```diff
@@ -0,0 +1,70 @@
+ARG NODE_VERSION=20
+ARG RUBY_VERSION=3.3.6
+
+# Front Build
+FROM node:$NODE_VERSION-alpine AS front_build
+
+WORKDIR /app
+
+COPY ./front/ .
+
+RUN apk add python3 build-base
+RUN yarn && yarn build && npm prune --omit=dev
+
+# API Build
+FROM ruby:$RUBY_VERSION-slim AS api_build
+
+ENV BUNDLER_VERSION='2.5.5'
+ENV PATH="$PATH:/root/.cargo/bin/"
+
+WORKDIR /app
+
+RUN apt-get update && apt-get upgrade -y && \
+    apt-get install nodejs curl build-essential git pkg-config libpq-dev libclang-dev curl -y && \
+    curl https://sh.rustup.rs -sSf | bash -s -- -y
+
+COPY ./api/Gemfile /app/Gemfile
+COPY ./api/Gemfile.lock /app/Gemfile.lock
+
+RUN gem install bundler --no-document -v '2.5.5' && \
+    gem install foreman && \
+    bundle config build.nokogiri --use-system-libraries &&\
+    bundle install --jobs=3 --retry=3 --without development test
+
+# Final Image
+FROM ruby:$RUBY_VERSION-slim
+
+WORKDIR /app
+
+RUN apt-get update && apt-get upgrade -y && \
+    apt-get install curl ca-certificates gnupg software-properties-common -y
+
+RUN curl -fsSL https://postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor | tee /usr/share/keyrings/postgresql-archive-keyring.gpg > /dev/null
```

Comment:
you will create less docker layers if you do 1 command, this also should reduce the size normally

Reply:
done!